raiden-network/light-client
# Feature/add disclaimer screen

Part of #1658 for the dApp

**Short description**

- add disclaimer route
- implement design for disclaimer view
- extend store for disclaimer related state
- introduce global navigation guard to protect access without accepting the disclaimer


**Definition of Done**


**Steps to manually test the change (dApp)**

_Test Cases_:
1. Go to the landing page, accept the disclaimer without clicking the second checkbox and continue -> navigation should work fine, starting with the connection screen
2. Do 1. but try to reload/directory access any page -> should redirect to disclaimer again (resets stores state)
3. Do 1. with a specific address like `/account`, accept disclaimer -> should redirect to account (eventually require to connect first)
4. Accept disclaimer and check to persist -> try again 2. and 3. which should no more redirect first to disclaimer
5. Do 4. (actually you have already) and try to access the landing page ('/') again -> should skip directly to connect screen or if already connected to home screen
